packages/claim-cli/src/applications/publish-multisig-claim.ts

```diff
@@ -0,0 +1,142 @@
+import { ethers, ZeroHash } from 'ethers';
+import { select } from '@inquirer/prompts';
+
+import L2ClaimAbi from '../abi/L2Claim';
+import {
+	NetworkParams,
+	fetchCheckEligibility,
+	getETHWallet,
+	confirmSendTransaction,
+	printPreview,
+	getInput,
+} from '../utils/';
+
+export default async function publishMultisigClaim(
+	networkParams: NetworkParams,
+	address: string | null = null,
+): Promise<void> {
+	const provider = new ethers.JsonRpcProvider(networkParams.rpc);
+	const lskAddress = address || (await getInput({ message: 'Multisig Address to be published' }));
+	const claimContract = new ethers.Contract(networkParams.l2Claim, L2ClaimAbi, provider);
+
+	const result = await fetchCheckEligibility(lskAddress, networkParams);
+	if (!result.account) {
+		console.log(`Address ${lskAddress} has no eligibility.`);
+		return process.exit(1);
+	}
+
+	if (!result.account.ready) {
+		console.log(`Address ${lskAddress} has insufficient signatures.`);
+		return process.exit(1);
+	}
+
+	const claimedTo = await claimContract.claimedTo(result.account.address);
```

Comment:
I think its better to create contract instance after all conditions are checked.
```suggestion
	const result = await fetchCheckEligibility(lskAddress, networkParams);
	if (!result.account) {
		console.log(`Address ${lskAddress} has no eligibility.`);
		return process.exit(1);
	}

	if (!result.account.ready) {
		console.log(`Address ${lskAddress} has insufficient signatures.`);
		return process.exit(1);
	}

	const claimContract = new ethers.Contract(networkParams.l2Claim, L2ClaimAbi, provider);
	const claimedTo = await claimContract.claimedTo(result.account.address);
```